testing/infra/terraform/modules/ec_deployment/outputs.tf

```
@@ -50,11 +50,11 @@ output "admin_console_url" {
 }
 
 output "apm_component_id" {
-  value       = ec_deployment.deployment.integrations_server[0].resource_id
+  value       = var.integrations_server ? ec_deployment.deployment.integrations_server[0].resource_id : ec_deployment.deployment.apm[0].resource_id
   description = "APM Cloud component ID"
 }
 
 output "fleet_component_id" {
-  value       = ec_deployment.deployment.integrations_server[0].resource_id
+  value       = var.integrations_server ? ec_deployment.deployment.integrations_server[0].resource_id : ""
   description = "Fleet Server Cloud component ID"
```

I'd suggest to mention that this field is expected to be empty in some cases:
```suggestion
  description = "Fleet Server Cloud component ID; empty for 7.x clusters"
```